django/contrib/gis/gdal/raster/source.py

```diff
@@ -151,6 +184,13 @@ def __init__(self, ds_input, write=False):
         else:
             raise GDALException('Invalid data source input type: "{}".'.format(type(ds_input)))
 
+    def __del__(self):
```

Comment:
Unfortunately I can't give input on that. I have never heard of `weakref` before, and I don't know enough about pythons gc etc to have an informed opinion.